File: source-connector/src/main/java/sourceconnector/service/offset/OffsetRecordServiceImpl.java
package sourceconnector.service.offset;

import lombok.RequiredArgsConstructor;
import offsetmanager.domain.file.FileKey;
import offsetmanager.domain.offset.OffsetRecord;
import sourceconnector.domain.connect.OffsetRecordService;

import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;

@RequiredArgsConstructor
public class OffsetRecordServiceImpl implements OffsetRecordService {
  private final OffsetRecordRepository offsetRecordRepository;

  @Override
  public Map<FileKey, Long> offsetMap(List<FileKey> fileKeys) {
    List<OffsetRecord> offsetRecords = this.offsetRecordRepository.findLastOffsetRecords(fileKeys);
    return offsetRecords.stream()
      .collect(Collectors.toMap(
        OffsetRecord::key,
        OffsetRecord::offset
      ));
  }
}

File: source-connector/src/main/kotlin/sourceconnector/service/offset/OffsetRecordServiceImpl.kt
package sourceconnector.service.offset

import offsetmanager.domain.file.FileKey
import offsetmanager.domain.offset.OffsetRecord
import sourceconnector.domain.connect.OffsetRecordService

class OffsetRecordServiceImpl(
  private val offsetRecordRepository: OffsetRecordRepository
) : OffsetRecordService {

  override fun offsetMap(fileKeys: List<FileKey>): Map<FileKey, Long> {
    val offsetRecords: List<OffsetRecord> = this.offsetRecordRepository.findLastOffsetRecords(fileKeys)
    return offsetRecords.associate { it.key() to it.offset() }
  }
}
